supernode/server/api/utils.go

```diff
@@ -52,26 +52,28 @@ func ParseJSONRequest(req io.Reader, target interface{}, validator ValidateFunc)
 	return nil
 }
 
-// EncodeResponse encodes response in json.
-// The response body is empty if the data is nil or empty value.
-func EncodeResponse(w http.ResponseWriter, code int, data interface{}) error {
+// SendResponse encodes response in json.
+//
+// TODO:
+//  Should the response body should be empty if the data is nil or empty
+//  string? Now it's incompatible with the client.
+func SendResponse(w http.ResponseWriter, code int, data interface{}) error {
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(code)
-	if util.IsNil(data) || data == "" {
-		return nil
-	}
-	return json.NewEncoder(w).Encode(data)
+	enc := json.NewEncoder(w)
+	enc.SetEscapeHTML(false)
+	return enc.Encode(data)
 }
 
 // HandleErrorResponse handles err from server side and constructs response
 // for client side.
 func HandleErrorResponse(w http.ResponseWriter, err error) {
 	switch e := err.(type) {
 	case *errortypes.HTTPError:
-		_ = EncodeResponse(w, e.Code, errResp(e.Code, e.Msg))
+		_ = SendResponse(w, e.Code, errResp(e.Code, e.Msg))
```

Comment:
This is odd, The error should not be ignored silently, right?

Reply:
done